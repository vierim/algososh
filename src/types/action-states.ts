export enum ActionStates {
  Waiting = '',
  AddToHead = 'AddToHead',
  AddToTail = 'AddToTail',
  DeleteFromHead = 'DeleteFromHead',
  DeleteFromTail = 'DeleteFromTail',
  AddByIndex = 'AddByIndex',
  DeleteByIndex = 'DeleteByIndex',
}
