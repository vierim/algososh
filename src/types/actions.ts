export enum Actions {
  Waiting = '',
  Push = 'Push',
  Pop = 'Pop',
}
