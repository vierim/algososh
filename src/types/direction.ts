export enum Direction {
  Ascending = 'Ascending',
  Descending = 'Descending',
}
