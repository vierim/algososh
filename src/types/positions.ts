export enum Positions {
  Top = 'Top',
  Bottom = 'Bottom',
}
