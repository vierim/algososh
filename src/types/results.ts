import { ElementStates } from './element-states';

export type TReverseRangeResult = {
  id: number;
  value: string;
  state: ElementStates;
}[];

export type TSortingResult = {
  array: number[];
  current: number[];
  modified: number[];
};
