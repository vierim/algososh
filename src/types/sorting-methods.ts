export enum SortingMethods {
  Bubble = 'Bubble',
  Choice = 'Choice',
}
