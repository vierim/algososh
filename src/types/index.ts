import { TSortingResult, TReverseStringResult } from "./results";

export type { TSortingResult, TReverseStringResult };
