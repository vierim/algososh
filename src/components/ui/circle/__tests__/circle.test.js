import renderer from 'react-test-renderer';
import { ElementStates } from '../../../../types';

import { Circle } from '../circle';

describe('Тестирование компонента Circle', () => {
  test('Рендер компонента без буквы', () => {
    const tree = renderer
      .create(<Circle />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Рендер компонента с буквами', () => {
    const tree = renderer
      .create(<Circle letter='123' />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Рендер компонента с head', () => {
    const tree = renderer
      .create(<Circle head={'head'} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Рендер компонента с React-элементом в head', () => {
    const tree = renderer
      .create(<Circle head={<div>ReactElement</div>} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // с tail;
  test('Рендер компонента с tail', () => {
    const tree = renderer
      .create(<Circle tail={'tail'} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // с react-элементом в tail;
  test('Рендер компонента с React-элементом в tail', () => {
    const tree = renderer
      .create(<Circle tail={<div>ReactElement</div>} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // с index;

  test('Рендер компонента с индексом', () => {
    const tree = renderer
      .create(<Circle index={1} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // с пропом isSmall ===  true;
  test('Рендер компонента isSmall ===  true', () => {
    const tree = renderer
      .create(<Circle isSmall={true} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // в состоянии default;
  test('Рендер компонента в состоянии default', () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  
  // в состоянии changing;
  test('Рендер компонента в состоянии changing', () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // в состоянии modified.
  test('Рендер компонента в состоянии modified', () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
