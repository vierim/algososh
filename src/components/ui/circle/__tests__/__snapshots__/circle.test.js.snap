// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle Рендер компонента isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента с React-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      ReactElement
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента с React-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div>
      ReactElement
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер компонента с индексом 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;
